tools/update-tester/src/testrunner.rs

```diff
@@ -179,99 +227,143 @@ struct TestClient(Client);
+    fn install_toolkit_at_version(&mut self, old_toolkit_version: &str) {
         let create = format!(
             r#"CREATE EXTENSION timescaledb_toolkit VERSION "{}""#,
-            old_version
+            old_toolkit_version
         );
         self.simple_query(&create).unwrap_or_else(|e| {
             panic!(
                 "could not install extension at version {} due to {}",
-                old_version, e,
+                old_toolkit_version, e,
             )
         });
     }
 
-    #[must_use]
-    fn create_test_objects(&mut self) -> QueryValues {
-        let create_data_table = "\
-            CREATE TABLE test_data(ts timestamptz, val DOUBLE PRECISION);\
-            INSERT INTO test_data \
-                SELECT '2020-01-01 00:00:00+00'::timestamptz + i * '1 hour'::interval, \
-                100 + i % 100\
-            FROM generate_series(0, 10000) i;\
-        ";
-        self.simple_query(create_data_table)
-            .unwrap_or_else(|e| panic!("could create the data table due to {}", e));
-
-        // TODO JOSH - I want to have additional stuff for newer versions,
-        //             but it's not ready yet
-        let create_test_view = "\
-            CREATE MATERIALIZED VIEW regression_view AS \
-                SELECT \
-                    counter_agg(ts, val) AS countagg, \
-                    hyperloglog(1024, val) AS hll, \
-                    time_weight('locf', ts, val) AS twa, \
-                    uddsketch(100, 0.001, val) as udd, \
-                    tdigest(100, val) as tdig, \
-                    stats_agg(val) as stats \
-                FROM test_data;\
-        ";
-        self.simple_query(create_test_view)
-            .unwrap_or_else(|e| panic!("could create the regression view due to {}", e));
-
-        let query_test_view = "\
-            SET TIME ZONE 'UTC'; \
-            SELECT \
-                num_resets(countagg), \
-                distinct_count(hll), \
-                average(twa), \
-                approx_percentile(0.1, udd), \
-                approx_percentile(0.1, tdig), \
-                kurtosis(stats) \
-            FROM regression_view;\
-        ";
-        let view_output = self
-            .simple_query(query_test_view)
-            .unwrap_or_else(|e| panic!("could query the regression view due to {}", e));
-        get_values(view_output)
+    fn set_timezone_utc(&mut self) {
+        self.simple_query("SET TIME ZONE 'UTC';")
+            .unwrap_or_else(|e| panic!("could not set time zone to UTC due to {}", e));
+    }
+
+    fn create_test_objects_from_file(
+        &mut self,
+        root_config: ConnectionConfig<'_>,
+        current_toolkit_version: String,
+    ) -> Vec<(Test, Result<(), TestError>)> {
+        let all_tests = parser::extract_tests("sql");
+        // Hack to match previous versions of toolkit that don't conform to Semver.
+        let current_toolkit_semver = match current_toolkit_version.as_str() {
```

Comment:
[After live discussion, we think I was confused when I wrote this, because actually we want the tests even for old versions to come FROM THE NEW CODE.  Not only would running old `create-test-objects` be pointless, but running new `create-test-objects` lets us add new tests for old versions.
So ignore what's below for now.
Recording for posterity.]

Is this necessary?  Two questions make me think it may not be:

1. I want to drop all releases older than 1.9 from `upgradeable_from` for 1.12 and all older than 1.10 for 1.13.
2. Will we be able to run 1.11's validate-test-objects against 1.10's create-test?  If not, s/1.10/1.11/ in #1 :)

I don't think I've spoken of my push to drop <1.9 from upgradeable_from so let me get down an early draft here:

We can't `create-test-objects` on any release older than 1.10, which means our current test automation plan won't support testing upgrading from any release older than that without having two automation schemes at twice the price (of both build AND maintain!), which means we can't call "release automation" done until we drop the old versions.